src/python/WMComponent/WorkflowUpdater/WorkflowUpdaterPoller.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python
+"""
+The WorkflowUpdater poller component.
+Among the actions performed by this component, we can list:
+* find active workflows in the agent
+* filter those that require pileup dataset
+* find out the current location for the pileup datasets
+* get a list of blocks available and locked by WM
+* match those blocks with the current pileup config json file. In other words,
+  blocks that are no longer locked and/or available need to be removed from the
+  json file.
+* update this json in the workflow sandbox
+"""
+
+import logging
+import os
+import threading
+
+from Utils.CertTools import cert, ckey
+from Utils.Timers import timeFunction
+from WMCore.Services.Rucio.Rucio import Rucio
+from WMCore.Services.pycurl_manager import RequestHandler
+from WMCore.WMException import WMException
+from WMCore.WMSpec.WMWorkload import WMWorkloadHelper
+from WMCore.WorkerThreads.BaseWorkerThread import BaseWorkerThread
+from WMCore.DAOFactory import DAOFactory
+
+
+class WorkflowUpdaterException(WMException):
+    """
+    Specific WorkflowUpdaterPoller exception handling.
+    """
+
+
+class WorkflowUpdaterPoller(BaseWorkerThread):
+    """
+    Poller that does the actual work for updating workflows.
+    """
+
+    def __init__(self, config):
+        """
+        Initialize WorkflowUpdaterPoller object
+        :param config: a Configuration object with the component configuration
+        """
+        super(BaseWorkerThread, self).__init__()
+
+        myThread = threading.currentThread()
+        self.daoFactory = DAOFactory(package="WMCore.WMBS",
+                                     logger=myThread.logger,
+                                     dbinterface=myThread.dbi)
+        self.listActiveWflows = self.daoFactory(classname="Workflow.GetUnfinishedWorkflows")
+
+        # configuration
+        self.config = config
+        self.rucioAcct = getattr(config.WorkflowUpdater, "rucioAccount", "wma_test")
+        self.rucioUrl = getattr(config.WorkflowUpdater, "rucioUrl",
+                                "http://cms-rucio-int.cern.ch")
```

Comment:
Are you real?

First, this is not hard-coded, this is what people call a "default value". There are no rules that say that a default value needs to be configurable and/or loaded through environment variable, to the best of my knowledge. Default values exist when we expect it to be overriden, but still want to make it functional even if not.

Second, you should consider the same concern when you contribute to the project as well:
https://github.com/dmwm/WMCore/blob/master/test/python/WMCore_t/MicroService_t/MSPileup_t/MSPileupData_t.py#L25 and I am sure in many other places.

Said that, I am happy to make these attributes mandatory by directly accessing it in the object. Python exception is good enough such that we can see exactly which attribute it misses, in case one is not provided.

Reply:
Alan, I'm human, and prove to the same mistakes :) When I review the code I try to be objective (and since I did work with CMSWEB internals I provide such comments). And, I appreciate that you point me to my mistakes and my review comments regardless how I may feel stupid and/or awkward afterwards. Said that,  my intention was to address my findings within concrete review.

Reply:
Despite making me upset to read this comment, I think it's a legit concern and I have materialized it in this GH issue: https://github.com/dmwm/WMCore/issues/11799 Feel free to add anything else that might be wrong or missing. Thanks